build/Helix/OutputTestResults.ps1

```diff
@@ -30,19 +31,23 @@ $totalTestsExecutedCount = 0
 
 foreach ($testRun in ($testRuns.value | Sort-Object -Property "completedDate"))
 {
-    if (-not $timesSeenByRunName.ContainsKey($testRun.name))
+    # The same build for a pull request can have multiple test runs associated with it if the build owner opted to re-run a test run.
+    # We should only pay attention to the current attempt version.
+    # NB: If in the future we have pull request builds do multiple test runs as part of the same build definition, we'll need to revisit this.
+    if ($BuildReason -ieq "PullRequest")
```

Comment:
Nit: instead of making it specific to pull requests - perhaps name it something like always use latest results or something similar so that it is not coupled with a particular pipeline. That way it is a behavior that is picked by the individual pipelines.

Reply:
That's not a bad idea - I'll do that instead.